geoh5py/workspace/workspace.py

```diff
@@ -439,12 +442,12 @@ def create_data(
 
     def create_entity(
         self,
-        entity_class,
+        entity_class: type[TYPE],
         compression: int = 5,
         entity: dict | None = None,
         entity_type: EntityType | dict | None = None,
         save_on_creation: bool = True,
-    ) -> Entity:
+    ) -> TYPE:
```

Comment:
Why the change?

Reply:
Because now the typing of this function will be the class type you sent it as attribute.
As there is no risk the type change (or it means an error is raised), it's super convenient!

Reply:
I just removed the typing